File: src/example/java/de/flauschig/eucalyptus/example/handler/ExampleListener.java
package de.flauschig.eucalyptus.example.handler;

import de.flauschig.eucalyptus.handler.EventHandler;
import de.flauschig.eucalyptus.handler.EventListener;
import de.flauschig.eucalyptus.example.events.ExampleEvent;

public class ExampleListener implements EventListener {
    @EventHandler
    public void exampleEvent(ExampleEvent exampleEvent) {
        System.out.println(exampleEvent.getText());
    }
}

File: src/example/kotlin/de/flauschig/eucalyptus/example/handler/ExampleListener.kt
package de.flauschig.eucalyptus.example.handler;

import de.flauschig.eucalyptus.handler.EventHandler
import de.flauschig.eucalyptus.handler.EventListener
import de.flauschig.eucalyptus.example.events.ExampleEvent

class ExampleListener : EventListener {
    @EventHandler
    fun exampleEvent(exampleEvent: ExampleEvent) {
        println(exampleEvent.text)
    }
}